src/pkg/metrics/metrics.go

```diff
@@ -12,33 +13,47 @@ type KindStats struct {
 	ResourceCount int64
 }
 
+var (
+	bytesRead count.Key = "bytes-read"
+	resources count.Key = "resources"
+)
+
 type ExportStats struct {
 	// data is kept private so that we can enforce atomic int updates
-	data map[path.CategoryType]KindStats
+	data syncd.MapOf[path.CategoryType, *count.Bus]
 }
 
-func (es *ExportStats) UpdateBytes(kind path.CategoryType, bytesRead int64) {
-	if es.data == nil {
-		es.data = map[path.CategoryType]KindStats{}
-	}
-
-	ks := es.data[kind]
-	atomic.AddInt64(&ks.BytesRead, bytesRead)
-	es.data[kind] = ks
+func (es *ExportStats) UpdateBytes(kind path.CategoryType, numBytes int64) {
+	es.getCB(kind).Add(bytesRead, numBytes)
 }
 
 func (es *ExportStats) UpdateResourceCount(kind path.CategoryType) {
-	if es.data == nil {
-		es.data = map[path.CategoryType]KindStats{}
+	es.getCB(kind).Inc(resources)
+}
+
+func (es *ExportStats) getCB(kind path.CategoryType) *count.Bus {
+	es.data.LazyInit()
+
+	cb, ok := es.data.Load(kind)
+	if !ok {
+		cb = count.New()
+		es.data.Store(kind, cb)
 	}
 
-	ks := es.data[kind]
-	atomic.AddInt64(&ks.ResourceCount, 1)
-	es.data[kind] = ks
+	return cb
 }
 
 func (es *ExportStats) GetStats() map[path.CategoryType]KindStats {
-	return es.data
+	toKindStats := map[path.CategoryType]KindStats{}
+
+	for k, cb := range es.data.Values() {
+		toKindStats[k] = KindStats{
+			BytesRead:     cb.Get(bytesRead),
+			ResourceCount: cb.Get(resources),
+		}
+	}
+
+	return toKindStats
```

Comment:
The current downstream implementations calls `GetStats` and pass around the map which would get updated only to pull the values out at this end. We should coordinate to ensure that we pass around `ExportStats` instead of the value got from `GetStats`.

Reply:
I don't see the map getting re-used.  `exportStats.GetStats` is called here: https://github.com/alcionai/corso/blob/main/src/internal/operations/export.go#L327.  That func is only called here in the CLI: https://github.com/alcionai/corso/blob/main/src/cli/export/export.go#L134, which prints, but doesn't update, the map.  Am I missing its usage somewhere?

Reply:
I meant the SDK consumers, not code within corso.

Reply:
I think that'll have to be fixed on the SDK side.  Those users can store the ExportStats as a `GetStatser` interface, and call `GetStats()` at the end of the operation.  That'll provide an equivalent usage.

Reply:
@zackrossman tagging you just so that you are aware of an upcoming change.